bcmc/src/test/java/com/adyen/checkout/bcmc/BcmcComponentParamsMapperTest.kt

```diff
@@ -100,29 +77,104 @@ internal class BcmcComponentParamsMapperTest {
+        showStorePaymentField: Boolean,
+        enableStoreDetails: Boolean,
+        isStorePaymentFieldVisible: Boolean
+    ) {
+        val bcmcConfiguration = getBcmcConfigurationBuilder()
+            .setShowStorePaymentField(showStorePaymentField)
+            .build()
+        val sessionSetupResponse =
+            getSessionSetupResponse(configuration = Configuration(enableStoreDetails = enableStoreDetails))
+
+        val params = BcmcComponentParamsMapper().mapToParams(
+            bcmcConfiguration,
+            sessionSetupResponse = sessionSetupResponse
+        )
+
+        val expected = getBcmcComponentParams(isStorePaymentFieldVisible = isStorePaymentFieldVisible)
+
+        assertEquals(expected, params)
+    }
+
+    private fun getSessionSetupResponse(
+        id: String = "id",
+        sessionData: String = "sessionData",
+        amount: Amount = Amount.EMPTY,
+        expiresAt: String = "expiresAt",
+        paymentMethods: PaymentMethodsApiResponse? = null,
+        returnUrl: String = "returnUrl",
+        configuration: Configuration
+    ): SessionSetupResponse = SessionSetupResponse(
+        id = id,
+        sessionData = sessionData,
+        amount = amount,
+        expiresAt = expiresAt,
+        paymentMethods = paymentMethods,
+        returnUrl = returnUrl,
+        configuration = configuration
+    )
+
+    private fun getBcmcConfigurationBuilder() = BcmcConfiguration.Builder(
+        shopperLocale = Locale.US,
+        environment = Environment.TEST,
+        clientKey = TEST_CLIENT_KEY_1
+    )
+
+    private fun getBcmcComponentParams(
+        shopperLocale: Locale = Locale.US,
+        environment: Environment = Environment.TEST,
+        clientKey: String = TEST_CLIENT_KEY_1,
+        isAnalyticsEnabled: Boolean = true,
+        isCreatedByDropIn: Boolean = false,
+        amount: Amount = Amount.EMPTY,
+        isSubmitButtonVisible: Boolean = true,
+        isHolderNameRequired: Boolean = false,
+        shopperReference: String? = null,
+        isStorePaymentFieldVisible: Boolean = false,
+    ) = BcmcComponentParams(
+        shopperLocale = shopperLocale,
+        environment = environment,
+        clientKey = clientKey,
+        isAnalyticsEnabled = isAnalyticsEnabled,
+        isCreatedByDropIn = isCreatedByDropIn,
+        amount = amount,
+        isSubmitButtonVisible = isSubmitButtonVisible,
+        isHolderNameRequired = isHolderNameRequired,
+        shopperReference = shopperReference,
+        isStorePaymentFieldVisible = isStorePaymentFieldVisible
+    )
+
     companion object {
         private const val TEST_CLIENT_KEY_1 = "test_qwertyuiopasdfghjklzxcvbnmqwerty"
         private const val TEST_CLIENT_KEY_2 = "live_qwertyui34566776787zxcvbnmqwerty"
+
+        @JvmStatic
+        fun sessionSetupConfigurationSource() = listOf(
+            arguments(false, false, false),
```

Comment:
This doesn't seem very readable, you can to check the main methods to know what each field is for. How about we create a data class with these 3 booleans and use it as a parameter for the test? This way we can used named arguments and we'll be able to read what each boolean is for

Reply:
As you said, the test method makes it clear what each boolean means. IMO we could also just add a little comment here to tell what each boolean means

Reply:
I agree with @OscarSpruit I guess just a comment should be enough.